http/transport/circuit_breaker_tripper_test.go

```diff
@@ -0,0 +1,64 @@
+package transport
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/sony/gobreaker"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCircuitBreakerTripper(t *testing.T) {
+	req := httptest.NewRequest("GET", "/foo", nil)
+
+	t.Run("with_default_settings", func(t *testing.T) {
+		breaker := NewDefaultCircuitBreakerTripper("testcircuitbreaker")
+		chain := Chain(breaker).Final(&failingRoundTripper{})
+
+		for i := 0; i < 6; i++ {
+			if _, err := chain.RoundTrip(req); errors.Is(err, gobreaker.ErrOpenState) {
+				t.Errorf("got err=%q, before expected", gobreaker.ErrOpenState)
+			}
+		}
+
+		if _, err := chain.RoundTrip(req); !errors.Is(err, gobreaker.ErrOpenState) {
+			t.Errorf("wanted err=%q, got err=%q", gobreaker.ErrOpenState, err)
+		}
+	})
+
+	t.Run("panic_on_empty_name", func(t *testing.T) {
+		assert.Panics(t, assert.PanicTestFunc(func() {
+			NewCircuitBreakerTripper(gobreaker.Settings{})
+		}), "NewCircuitBreakerTripper did not panic on empty name")
+
+		assert.Panics(t, assert.PanicTestFunc(func() {
+			NewDefaultCircuitBreakerTripper("")
+		}), "NewDefaultCircuitBreakerTripper did not panic on empty name")
+
+	})
+
+	t.Run("resp_object_untouched", func(t *testing.T) {
+		wantBodyStr := "some text in the body which should be unaffected by the circuit breaker"
+		breaker := NewDefaultCircuitBreakerTripper("testcircuitbreaker")
+		chain := Chain(breaker).Final(&transportWithBody{body: wantBodyStr})
+
+		resp, err := chain.RoundTrip(req)
+		assert.NoError(t, err, "expected no err, got err=%q", err)
```

Comment:
```suggestion
		require.NoError(t, err, "expected no err, got err=%q", err)
```

`import "github.com/stretchr/testify/require"`

`require` prints the error and aborts the test while `assert` prints the error only. So use `require` if the following conditions will inevitable fail because the current failed.